base/runtime_internals.jl

```diff
@@ -1182,6 +1182,37 @@ function signature_type(@nospecialize(f), @nospecialize(argtypes))
     return rewrap_unionall(Tuple{ft, u.parameters...}, argtypes)
 end
 
+"""
+    delete_method(m::Method)
+
+Make method `m` uncallable and force recompilation of any methods that use(d) it.
+"""
+function delete_method(m::Method)
+    ccall(:jl_method_table_disable, Cvoid, (Any, Any), get_methodtable(m), m)
+end
+
+"""
+    freeze!(mt::Core.MethodTable)
+
+Disallow adding or modifying methods of `mt`.
+
+See also [`unfreeze!(mt)`](@ref unfreeze!).
+"""
+function freeze!(mt::Core.MethodTable)
+    ccall(:jl_method_table_set_frozen, Cvoid, (Any, Cint), mt, 1)
+end
+
+"""
+    unfreeze!(mt::Core.MethodTable)
+
+Allow adding or modifying methods of `mt`.
+
+See also [`freeze!(mt)`](@ref freeze!).
+"""
+function unfreeze!(mt::Core.MethodTable)
+    ccall(:jl_method_table_set_frozen, Cvoid, (Any, Cint), mt, 0)
+end
+
```

Comment:
I am not sure unfreeze is a good idea. If the function should be frozen, then don't freeze it.
```suggestion
```